docs/status.md

```diff
@@ -45,7 +45,7 @@ one of the following tracking labels.
 | abs                      | yes           | yes          | yes            | yes             | no          |
 | add                      | yes           | yes          | yes            | yes             | yes         |
 | after_all                | yes           | yes          | no             | yes             | no          |
-| all_gather               | no            | revisit      | no             | no              | no          |
+| all_gather               | yes           | yes          | infeasible     | no              | no          |
```

Comment:
Verification should be "revisit" because we aren't yet verifying `replica_groups` against `num_replicas` and `num_partitions`.

Reply:
I agree with that. 

As yopu pointed out, we do not have the [check](https://github.com/tensorflow/tensorflow/blob/20c6943d3cd7e07da162f7778a0af5d3776274b4/tensorflow/compiler/xla/service/hlo_verifier.cc#L436) for  `shard_count == subgroup_size`. The `subgroup_size` depends on [module configuration](https://github.com/tensorflow/tensorflow/blob/20c6943d3cd7e07da162f7778a0af5d3776274b4/tensorflow/compiler/xla/service/hlo_verifier.cc#L95)  which manages the settings and values which affect the compiled executable outside of the HLO code itself and I am not sure if that info is available in StableHLO IR.

Reply:
IMO we should have a ticket for exposing `num_replicas` and `num_partitions` in StableHLO to enable the check.